zbench.zig

```diff
@@ -183,117 +244,88 @@ pub const Benchmark = struct {
 
         return avg;
     }
-};
-
-/// BenchFunc is a function type that represents a benchmark function.
-/// It takes a pointer to a Benchmark object.
-pub const BenchFunc = fn (*Benchmark) void;
-
-/// BenchmarkResult stores the result of a single benchmark.
-/// It includes the name and the total duration of the benchmark.
-pub const BenchmarkResult = struct {
-    /// Name of the benchmark.
-    name: []const u8,
-    /// Total duration of the benchmark in nanoseconds.
-    duration: u64,
-};
-
-/// BenchmarkResults acts as a container for multiple benchmark results.
-/// It provides functionality to format and print these results.
-pub const BenchmarkResults = struct {
-    /// A dynamic list of BenchmarkResult objects.
-    results: std.ArrayList(BenchmarkResult),
-
-    /// Determines the color representation based on the duration of the benchmark.
-    /// duration: The duration to evaluate.
-    pub fn getColor(self: *const BenchmarkResults, duration: u64) c.Color {
-        const max_duration = @max(self.results.items[0].duration, self.results.items[self.results.items.len - 1].duration);
-        const min_duration = @min(self.results.items[0].duration, self.results.items[self.results.items.len - 1].duration);
-
-        if (duration <= min_duration) return c.Color.green;
-        if (duration >= max_duration) return c.Color.red;
-
-        const prop = (duration - min_duration) * 100 / (max_duration - min_duration + 1);
 
-        if (prop < 50) return c.Color.green;
-        if (prop < 75) return c.Color.yellow;
+    /// Calculate the standard deviation of the durations
+    // NOTE: We are doing integer division, roots and unsafe casting..
+    // Atleast it's very unlikely to overflow here, as this would
+    // only happen if the duration variance is over 514 years, and it's reasonable
+    // to assume no user will ever sit around and wait that for such a benchmark
+    // to complete, to then be able to calculate its standard deviation..
+    pub fn calculateStd(self: Benchmark) u64 {
+        if (self.durations.items.len <= 1) return 0;
+
+        const avg = self.calculateAverage();
+        var nvar: u64 = 0;
+        for (self.durations.items) |dur| {
+            const d: i64 = @bitCast(dur);
+            const a: i64 = @bitCast(avg);
+
+            nvar += @bitCast((d - a) * (d - a));
+        }
```

Comment:
the addition of the standard deviation is very good. makes me wonder why there is no stddev in std.math...